DEBUGGING.md

```diff
@@ -0,0 +1,45 @@
+# Debugging NUnit
```

Comment:
I love that you've added this file! 😄 One question, do you think it maybe belongs in the docs repo perhaps, rather than in the main repo root? Two reasons:
1. Root folder size.
2. This can be made generic if this goes well, and we add the same feature to the console.

Reply:
I would love to! Should that be done before or after this PR?

Reply:
Sorry - I meant in the docs **instead** of the nunit repo.

We only have key contribution docs at the moment in the repo root. This is a great doc - but I think it should be linked from the readme to docs, rather than living in the repo root itself - I don't see it as being on the same level as README, BUILDING and CONTRIBUTING. (Although a link from contributing as well might be a good idea!)

Reply:
I think I understood– you meant, move it to https://github.com/nunit/docs/wiki? Should I wait till this PR is approved and then move it, or let this PR be merged, then move to the wiki, then remove from this repo with a second PR?

Reply:
Ah, sorry - I thought you'd misunderstood, as I didn't see why you'd want to merge it, and then remove it! But I guess you're thinking of review purposes. 😄

Whatever's easier. Consider the doc reviewed by me! 👍